src/main/java/ch/njol/skript/expressions/ExprDustedStage.java

```diff
@@ -0,0 +1,91 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.Brushable;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Dusted Stage")
+@Description({
+	"Represents how far the block has been uncovered.",
+	"The only blocks that can currently be \"dusted\" are Suspicious Gravel and Suspicious Sand."
+})
+@Examples({
+	"send target block's maximum dusted stage",
+	"set {_sand}'s dusted stage to 2"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.20+")
+public class ExprDustedStage extends PropertyExpression<Block, Integer> {
+
+	private static final boolean SUPPORTS_DUSTING = Skript.classExists("org.bukkit.block.data.Brushable");
+
+	static {
+		if (SUPPORTS_DUSTING)
+			register(ExprDustedStage.class, Integer.class,
+				"[:max[imum]] dust[ed|ing] (value|stage|progress[ion])",
+				"blocks");
+	}
+
+	private boolean isMax;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Block>) exprs[0]);
+		isMax = parseResult.hasTag("max");
+		return true;
+	}
+
+	@Override
+	protected Integer[] get(Event event, Block[] source) {
+		return get(source, block -> {
+			if (block != null && block.getBlockData() instanceof Brushable brushable) {
+				return isMax ? brushable.getMaximumDusted() : brushable.getDusted();
+			}
+			return null;
+		});
+	}
+
+	@Override
+	public @Nullable Class<?>[] acceptChange(ChangeMode mode) {
+		if (!isMax && mode == Changer.ChangeMode.SET) {
+			return new Class[]{Integer.class};
+		}
+		return null;
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (isMax || mode != Changer.ChangeMode.SET || delta == null || delta.length == 0)
```

Comment:
```suggestion
		if (isMax || mode != ChangeMode.SET || delta == null || delta.length == 0)
```
delta won't be null and it'll always be set again here